### DISCUSSION TITLE
Why support from Safari 14.1?

### DISCUSSION THREAD

**Discussion by johnvanhulsen** [easepick/easepick#48] [Q&A]
Hi,

Can you explain why this package is only supported in Safari 14.1 and above?
Does it not work on older versions, or is it flex-gap or some css element that's missed in older versions?

Thanks for your great work!

**Comment by wakirin** [comment]
Hi,
As far as I remember, Safari below 14.1 does not support some JS features. (Unfortunately I don't remember exactly which features)
I've decided to drop support for older versions of Safari because Apple provides updates to Macs for at least 5 years.
If someone needs support for older browsers, they can use [Litepicker](https://github.com/wakirin/Litepicker).

**Comment by Muffinman** [comment]
This issue just hit me pretty late on in a project. Getting the following error in Safari 14.4.

```
SyntaxError: Unexpected token '='. Expected an opening '(' before a method's parameter list.
(anonymous function) — vendor.js:49
```

I agree that users should be upgrading to their latest available versions, but 14.4 is only 18 months old, it's a shame it's not supported.

The error is coming from https://github.com/easepick/easepick/blob/master/packages/datetime/src/index.ts but I'm not sure exactly which line.

**Reply by wakirin** [reply]
<img width="1013" alt="Screen Shot 2022-07-15 at 14 52 51" src="https://user-images.githubusercontent.com/7056055/179189423-6b43b916-656f-4bbf-9845-385484903805.png">

Unfortunately I don't have ability to check on 14.4 version, but it works on Safari 14.1, so I think 14.4 should support it too. 
Try to check your project dependencies or webpack configuration.

**Reply by Muffinman** [reply]
Sorry I was wrong about the version, our issue is on *Safari 14.0.3* on *iOS 14.4*.

Still, this was released [Jan 2021](https://en.wikipedia.org/wiki/Safari_version_history#Safari_14_2), so not exactly ancient.

**Reply by hajo-travanto** [reply]
Some important object oriented javascript features (e.g. static / public / private declarations) made it into 14.1 while they are missing in 14.0.x and below. I'm showing them a "update your browser" error message since analytics showed it only affects less then 0.2% of all users, but don't know about your userbase. In my case a fallback or workaround wouldn't be worth it now and never in the long run.

Besides that all users that get 14.0 should be able to update to 14.1 or newer 14.x releases via the appstore.
